runtime/dict.go

```diff
@@ -390,7 +390,8 @@ func (d *Dict) Len() int {
 
 // putItem associates value with key in d, returning the old associated value if
 // the key was added, or nil if it was not already present in d.
-func (d *Dict) putItem(f *Frame, key, value *Object) (*Object, *BaseException) {
+// fun is defining the operation on d, which can be add or delete the entry.
+func (d *Dict) putItem(f *Frame, key, value *Object, fun func(f *Frame, d *Dict, entry *dictEntry, hash *Int, key, value *Object, index int) *BaseException) (*Object, *BaseException) {
```

Comment:
yes, I didn't really like that neither, that's a lot of parameters.
totally agree with `overwrite` parameter, at least we can make setdefault() cleaner, and it's not gonna mess up putitem() anyway.
let me revert this.